packages/expo-navigation-bar/src/NavigationBar.ts

```diff
@@ -1,133 +1,39 @@
-import { Platform, type EventSubscription, UnavailabilityError } from 'expo-modules-core';
-import { useEffect, useState } from 'react';
-import { processColor } from 'react-native';
+import { Platform } from 'react-native';
 import { isEdgeToEdge } from 'react-native-is-edge-to-edge';
 
-import ExpoNavigationBar from './ExpoNavigationBar';
-import {
-  NavigationBarButtonStyle,
-  NavigationBarBehavior,
-  NavigationBarPosition,
-  NavigationBarVisibility,
-  NavigationBarVisibilityEvent,
-} from './NavigationBar.types';
+import { setVisibilityAsync as originalSetVisibilityAsync } from './NativeNavigationBarWrapper';
+import { NavigationBarStyle, NavigationBarVisibility } from './NavigationBar.types';
 
-let shouldWarnAboutEdgeToEdge = Platform.OS === 'android' && isEdgeToEdge();
+export * from './NavigationBar.types';
 
-function potentiallyWarnAboutEdgeToEdge() {
-  if (shouldWarnAboutEdgeToEdge) {
-    shouldWarnAboutEdgeToEdge = false; // warn once
-    console.warn(
-      'Using expo-navigation-bar in apps with edge-to-edge layout enabled may cause unexpected behavior. Instead, use the SystemBars component from react-native-edge-to-edge. Learn more: https://expo.fyi/edge-to-edge-system-bars'
-    );
-  }
-}
+// This line only imports the type information for TypeScript type checking.  It
+// doesn't import the actual module in the compiled JavaScript code.  The actual
+// module is imported conditionally with require() below, in order to avoid
+// importing the module if edge-to-edge is not enabled (which could throw if
+// it's not linked).
```

Comment:
```suggestion
// This line only imports the type information for TypeScript type checking.
// It doesn't import the actual module in the compiled JavaScript code. The actual
// module is imported conditionally with require() below, in order to avoid
// importing the module if edge-to-edge is not enabled (which could throw if
// it's not linked).
```